lib/ipmi_fru.c

```diff
@@ -4797,6 +4797,32 @@ f_type, uint8_t f_index, char *f_string)
 	/* Convert index from character to decimal */
 	f_index= f_index - 0x30;
 
+	if (f_type == 'b' && f_index == 9)		// use index '9' for editing 'board mfg date'
+	{
+		time_t total_secs;
+		uint32_t *board_mfg_date=&fru_data[3];
+
+		total_secs = ipmi_strptime("%Y-%m-%d %H:%M:%S", f_string);
+		ipmi_time2fru(total_secs, board_mfg_date);
```

Comment:
looks like a range check is required here. we don't want values that exceed 24 bits, but `ipmi_strptime` is capable of returning greater values.

Reply:
How should we check the range of the total_secs? 
The total_secs will be subtracted by SECS_FROM_1970_1996 and divided by 60 to get the minutes and mask down to three bytes. I think this is safe here.